### YamlMime:ManagedReference
items:
- uid: system.fabric._retry_async
  id: _retry_async
  parent: system.fabric
  children:
  - system.fabric._retry_async.executeWithRetry(Supplier<CompletableFuture<TResult>>,Function<Throwable, Boolean>,Function<Throwable, Throwable>)
  - system.fabric._retry_async.handleException(Throwable)
  - system.fabric._retry_async.RetryAsync(int,long)
  - system.fabric._retry_async.RetryAsync(long)
  href: system.fabric._retry_async.yml
  langs:
  - java
  name: RetryAsync<TResult>
  nameWithType: RetryAsync<TResult>
  fullName: system.fabric.RetryAsync<TResult>
  type: Class
  source:
    remote: &o0
      path: src/prod/src/java/system.fabric/src/system/fabric/RetryAsync.java
      branch: master
      repo: https://msazure.visualstudio.com/One/_git/sf-java
    path: src/prod/src/java/system.fabric/src/system/fabric/RetryAsync.java
    startLine: 10
  package: system.fabric
  syntax:
    content: public class RetryAsync<TResult>
    typeParameters:
    - type: 5f1bcb0c
  inheritance:
  - java.lang.Object
- uid: system.fabric._retry_async.executeWithRetry(Supplier<CompletableFuture<TResult>>,Function<Throwable, Boolean>,Function<Throwable, Throwable>)
  id: executeWithRetry(Supplier<CompletableFuture<TResult>>,Function<Throwable, Boolean>,Function<Throwable, Throwable>)
  parent: system.fabric._retry_async
  href: system.fabric._retry_async.yml
  langs:
  - java
  name: executeWithRetry(Supplier<CompletableFuture<TResult>> task, Function<Throwable, Boolean> doRetry, Function<Throwable, Throwable> getRetryException)
  nameWithType: RetryAsync<TResult>.executeWithRetry(Supplier<CompletableFuture<TResult>> task, Function<Throwable, Boolean> doRetry, Function<Throwable, Throwable> getRetryException)
  fullName: CompletableFuture<TResult> system.fabric.RetryAsync<TResult>.executeWithRetry(Supplier<CompletableFuture<TResult>> task, Function<Throwable, Boolean> doRetry, Function<Throwable, Throwable> getRetryException)
  overload: system.fabric._retry_async.executeWithRetry*
  type: Method
  source:
    remote: *o0
    path: src/prod/src/java/system.fabric/src/system/fabric/RetryAsync.java
    startLine: 35
  package: system.fabric
  syntax:
    content: public CompletableFuture<TResult> executeWithRetry(Supplier<CompletableFuture<TResult>> task, Function<Throwable, Boolean> doRetry, Function<Throwable, Throwable> getRetryException)
    parameters:
    - id: task
      type: 99177cbd
    - id: doRetry
      type: 8c3f0c17
    - id: getRetryException
      type: 98c9f4fd
    return:
      type: 91cbf258
- uid: system.fabric._retry_async.handleException(Throwable)
  id: handleException(Throwable)
  parent: system.fabric._retry_async
  href: system.fabric._retry_async.yml
  langs:
  - java
  name: handleException(Throwable ex)
  nameWithType: RetryAsync<TResult>.handleException(Throwable ex)
  fullName: void system.fabric.RetryAsync<TResult>.handleException(Throwable ex)
  overload: system.fabric._retry_async.handleException*
  type: Method
  source:
    remote: *o0
    path: src/prod/src/java/system.fabric/src/system/fabric/RetryAsync.java
    startLine: 80
  package: system.fabric
  syntax:
    content: public void handleException(Throwable ex)
    parameters:
    - id: ex
      type: a04fafd7
- uid: system.fabric._retry_async.RetryAsync(int,long)
  id: RetryAsync(int,long)
  parent: system.fabric._retry_async
  href: system.fabric._retry_async.yml
  langs:
  - java
  name: RetryAsync(int maxTries, long delay)
  nameWithType: RetryAsync<TResult>.RetryAsync(int maxTries, long delay)
  fullName: system.fabric.RetryAsync<TResult>.RetryAsync(int maxTries, long delay)
  overload: system.fabric._retry_async.RetryAsync*
  type: Constructor
  source:
    remote: *o0
    path: src/prod/src/java/system.fabric/src/system/fabric/RetryAsync.java
    startLine: 21
  package: system.fabric
  syntax:
    content: public RetryAsync(int maxTries, long delay)
    parameters:
    - id: maxTries
      type: f75371fa
    - id: delay
      type: 0364520f
- uid: system.fabric._retry_async.RetryAsync(long)
  id: RetryAsync(long)
  parent: system.fabric._retry_async
  href: system.fabric._retry_async.yml
  langs:
  - java
  name: RetryAsync(long delay)
  nameWithType: RetryAsync<TResult>.RetryAsync(long delay)
  fullName: system.fabric.RetryAsync<TResult>.RetryAsync(long delay)
  overload: system.fabric._retry_async.RetryAsync*
  type: Constructor
  source:
    remote: *o0
    path: src/prod/src/java/system.fabric/src/system/fabric/RetryAsync.java
    startLine: 28
  package: system.fabric
  syntax:
    content: public RetryAsync(long delay)
    parameters:
    - id: delay
      type: 0364520f
references:
- uid: 5f1bcb0c
  spec.java:
  - name: TResult
    fullName: TResult
- uid: f75371fa
  spec.java:
  - name: int
    fullName: int
- uid: 0364520f
  spec.java:
  - name: long
    fullName: long
- uid: system.fabric._retry_async.RetryAsync*
  name: RetryAsync
  nameWithType: RetryAsync<TResult>.RetryAsync
  fullName: system.fabric.RetryAsync<TResult>.RetryAsync
  package: system.fabric
- uid: 99177cbd
  spec.java:
  - name: Supplier<CompletableFuture<TResult>>
    fullName: Supplier<CompletableFuture<TResult>>
- uid: 8c3f0c17
  spec.java:
  - name: Function<Throwable, Boolean>
    fullName: Function<Throwable, Boolean>
- uid: 98c9f4fd
  spec.java:
  - name: Function<Throwable, Throwable>
    fullName: Function<Throwable, Throwable>
- uid: 91cbf258
  spec.java:
  - name: CompletableFuture<TResult>
    fullName: CompletableFuture<TResult>
- uid: system.fabric._retry_async.executeWithRetry*
  name: executeWithRetry
  nameWithType: RetryAsync<TResult>.executeWithRetry
  fullName: CompletableFuture<TResult> system.fabric.RetryAsync<TResult>.executeWithRetry
  package: system.fabric
- uid: a04fafd7
  spec.java:
  - name: Throwable
    fullName: Throwable
- uid: system.fabric._retry_async.handleException*
  name: handleException
  nameWithType: RetryAsync<TResult>.handleException
  fullName: void system.fabric.RetryAsync<TResult>.handleException
  package: system.fabric
